src/api/c/approx.cpp

```diff
@@ -86,24 +86,29 @@ af_err af_approx1_uniform(af_array *yo, const af_array yi,
             return af_create_handle(yo, 0, nullptr, yi_info.getType());
         }
 
-        af_array output;
+        dim4 yo_dims = yi_dims;
+        yo_dims[xdim] = xo_dims[xdim];
+        if (*yo == 0) {
+            af_create_handle(yo, yo_dims.ndims(), yo_dims.get(), yi_info.getType());
```

Comment:
We should try to avoid using C-API at C-API level. In this case, the following will do.

```
*yo =  createHandle(yo_dims, yi_info.getType());
```

Reply:
Yeah that makes sense. Thanks for pointing it out!